backend/backend.go

```diff
@@ -296,6 +297,39 @@ func (backend *Backend) Config() *config.Config {
 	return backend.config
 }
 
+// Authenticate executes a system authentication. If authentication is disabled in
+// the config, the authentication is automatically successful.
+func (backend *Backend) Authenticate() {
+	backend.log.Info("Auth requested")
+	if backend.config.AppConfig().Backend.Authentication {
+		backend.environment.Auth()
+	} else {
+		backend.AuthResult(true)
+	}
+}
+
+// TriggerAuth triggers an auth-required notification.
+func (backend *Backend) TriggerAuth() {
+	backend.Notify(observable.Event{
+		Subject: "auth-required",
```

Comment:
imo it would be clearer to use the "auth" subject for all three events and use the Data field instead to send additional info which will be captured in the frontend subscription callback param. See for example

https://github.com/digitalbitbox/bitbox-wallet-app/blob/6b8ec3cc4b6a1037ffa93e19fb92d9cf5e731ff9/frontends/web/src/api/backend.ts#L79-L102 and https://github.com/digitalbitbox/bitbox-wallet-app/blob/6b8ec3cc4b6a1037ffa93e19fb92d9cf5e731ff9/backend/accounts.go#L571-L578

Reply:
Good point, done.